### ISSUE TITLE
Convert melodies to ABC notation?

### ISSUE THREAD

**Issue by mrZerial** [ricardomatias/playa#37]
Hello there. 
Do you think it'd be possible to add a function to convert whatever melodies come out to [AbcNotation](https://en.wikipedia.org/wiki/ABC_notation)?
I'm a musician and I've found your library really cool, generating random melodies and progressions, and I just think it'd be even cooler if we could get an abcNotation to toss in a abc-> sheet generator and get cool sheets out of randomized progressions. [Tonal](https://github.com/tonaljs/tonal/tree/master/packages/pitch-notation-abc) has a way apparently, but I'm no programmer and I've found it really hard trying to understand how to work out the timings from the melody instance so that I can add them to the notes and then use Tonal's function to generate the abcNotation. If not the full thing, maybe we could get an additional key to the Melody instance, next to (time,dur,next,midi,note) maybe a classic timing like '4n' ?

Anyways enough rambling, sorry if this isn't the right place for this but I just had to reach somehow. Cool stuff and keep it up

**Comment by ricardomatias** [comment]
Hey @mrZerial , thanks for the kind words. Do you want to convert them to music sheets or some specific AbcNotation format? Need to understand better what's your use case. Nevertheless it seems you can use `AbcNotation.scientificToAbcNotation("C#4");` to convert to Abc notation.

You can make use of [Time](https://ricardomatias.net/playa/?api#Core#Time) to convert between different formats, f.ex: `new Time(dur).notevalue` would convert to `4n` type of notation.

**Comment by mrZerial** [comment]
Okay after looking some things up a little more, I might have been a little unclear and/or wrong in the original post, so let me rephrase some things.

>  Do you want to convert them to music sheets or some specific AbcNotation format?

Music sheets is what my end goal is. I have a software I've been using that takes ABC notation and converts it to a music sheet [(also maybe this ?)](https://www.abcjs.net/). But maybe there's an easier way to do that without the need to convert to a different notation?

> Nevertheless it seems you can use AbcNotation.scientificToAbcNotation("C#4"); to convert to Abc notation.

I can but I'm not sure how to work out the timing, cause after getting the note out of that, say _C#4_ becomes _^C_, I also need to get C#4's timing to abc notation, otherwise the sheet will make no sense. Same goes for bars, and it'd be nice to also work out repeat sections and thick lines and what not but that's not really important right now.

> You can make use of Time to convert between different formats, f.ex: new Time(dur).notevalue would convert to 4n type of notation.

It looks like that might be the answer. I could swear I looked up and down through the Time section, I didn't notice that. 
So then this: 
`melody.map(x=> AbcNotation.scientificToAbcNotation(x.note) + new Time(x.dur).notevalue)`
gets me an array like this: 
`["F4nd", "C2n", ... ,"_D2nd", "_E4nd"]` 
so maybe I can then map the notevalue to a dictionary of abc lengths and compare with default **L** and assign if nessessary. Then I could have it count lengths and add bar lines, keeping it simple 4/4. Yeah, I mean, I think in the proccess of writing this comment we might have figured it out! I'll go try some stuff out. Thanks!!
